### PR TITLE
add a mechanism to only test a specific version

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
usage: --test --version <N>
 where N is the version number

### CODE REVIEW

File: src/main/main.cpp [hunk 1]
```diff
@@ -184,6 +184,9 @@ usage(int err = 1)
           "                           (Default is STELLAR_NETWORK_ID "
           "environment variable)\n"
           "      --test               Run self-tests\n"
+          "           --all-versions  Run with all possible protocol versions\n"
```
Comment (jonjove) [inline src/main/main.cpp]: I don't think we need to have `--all-versions` or `--version` documented here. They will appear in `--test -?` now. The documentation added here should be specified where they are added to the CLI.

Comment (MonsieurNicolas) [reply src/main/main.cpp]: ah yes the new version of Clara makes this irrelevant

**Comment by MonsieurNicolas** [comment]
@jonjove : I removed the changes that were in main.cpp

**Comment by jonjove** [comment]
r+ e82b9d8

**Pull request merged**